content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+
+There are two types of reader conditionals, standard and splicing. The standard conditional reader behaves similarly to a traditional `cond`. The syntax for usage is `#?` and looks like:
+
+[source,clojure]
+----
+#?(:clj  (Clojure expression)
+   :cljs (ClojureScript expression)
+   :clr  (Clojure CLR expression))
+----
+
+The syntax for a splicing conditional read is `#?@`. It is used to splice lists into the containing form. So the Clojure reader would read this:
+
+[source,clojure]
+----
+(defn build-list []
+  (list #?@(:clj  [5 6 7 8]
+            :cljs [1 2 3 4])))
+----
+
+as this:
+
+[source,clojure]
+----
+(defn build-list []
+  (list 5 6 7 8))
+----
+
+One important thing to note is that in Clojure 1.7 a splicing conditional reader cannot be used to splice in multiple top level forms. In concrete terms, this means you can't do this:
+
+[source,clojure]
+----
+;; Don't do this!, will throw an error
+#?@(:clj
+    [(defn clj-fn1 [] :abc)
+     (defn clj-fn2 [] :cde)])
+;; CompilerException java.lang.RuntimeException: Reader conditional splicing not allowed at the top level.
+----
+
+Instead you'd need to do wrap each function individually:
+
+[source,clojure]
+----
+#?(:clj (defn clj-fn1 [] :abc))
+#?(:clj (defn clj-fn2 [] :cde))
+----
+
+or use a `do` to wrap all of the top level functions:
+
+[source,clojure]
+----
+#?(:clj
+    (do (defn clj-fn1 [] :abc)
+        (defn clj-fn2 [] :cde)))
+----
+
+Which one you choose would probably depend on aesthetics, and what your plans for porting these functions to more Clojure platforms were. Let's go through some examples of places you might want to use these new reader conditionals.
+
+== Host interop
+
+Host interop is one of the biggest pain points that cljx and reader conditionals both solve. You may have a Clojure file that is almost pure Clojure, but needs to call out to the host environment for one function. https://github.com/lymingtonprecision/route-ccrs/blob/c579aea05504736f2cfbd31c3c755f7e25fdad77/src/route_ccrs/manufacturing_methods.cljc#L8-L10[This] is a classic example:
+
+[source,clojure]
+----
+(defn str->int [s]
+  #?(:clj  (java.lang.Integer/parseInt s)
+     :cljs (js/parseInt s)))
+----
+
+== Namespaces
+
+Namespaces are the other big pain point for sharing code between Clojure and ClojureScript. ClojureScript has different syntax for https://github.com/clojure/clojurescript/wiki/Differences-from-Clojure#lisp[requiring macros] than Clojure. To use these macros in a `.cljc` file, you'll need Reader Conditionals in the namespace declaration.
+
+https://github.com/VesaKarvonen[Vesa Karvonen] has a https://github.com/VesaKarvonen/poc.cml/blob/6c80c55aa16ab1253ad3e85705b5360fbed73f7e/src/poc/cml/sem.cljc[neat trick] for this, taking advantage of the fact you can have multiple `:require`'s in one Clojure ns form. **N.B.: ClojureScript ns forms can only have one `:require` in them.**
+
+[source,clojure]
+----
```

Comment:
Two comments on the examples:

1. Is it possible to simplify the second and third examples a bit? There's quite a bit of irrelevant code that obscures the point of the example somewhat (e.g. the let binding).
1. Would it make sense reverse the examples? Vesa's trick seems like an optimization best introduced last.